tools/scripts/clang_tidy_check.sh

```diff
@@ -31,7 +31,7 @@ fail() {
     exit 1
 }
 
-CLANG_TIDY_VERSION=12
+CLANG_TIDY_VERSION=15
```

Comment:
This script is also used for the git hooks and disables local checks when the user does not use a non-released clang-tidy. This is okay for the CI but for the hooks I would keep the current version.

That said, we could just print a warning and suggest to install clang-tidy 15